code/src/pages/not-found/NotFoundPage.test.tsx

```diff
@@ -0,0 +1,13 @@
+import { screen } from "@testing-library/react";
+import NotFoundPage from "@/pages/not-found/NotFoundPage";
+import { renderWithProviders } from "@/utils/test-utils";
+
+describe("NotFoundPage", () => {
+  test("renders the 404 title correctly", () => {
+    renderWithProviders(<NotFoundPage />);
+
+    const errorCode = screen.getByText("404");

```

Comment:
I suggest also test for heading existance on the page, what do you think? In my opinion, not found page should always have `h1` tag

Reply:
> I suggest also test for heading existance on the page, what do you think? In my opinion, not found page should always have `h1` tag

What do you mean ? It's a single test to cover the component which render `404` text. It will be enough
In one case you told too many tests and useless to tests even for each props. For now it's too less tests and should write else. Not understand you. It's test related to `NotFound` component which return `404` code

Reply:
It is semantically correct to have at least 1 `h1` tag per page

Reply:
> It is semantically correct to have at least one `h1` tag per page

Agree with you. Look at `variant`
<img width="550" alt="image" src="https://github.com/idx-academy/spa-orders/assets/56305508/4e23a65a-813b-421d-a41d-271c48ff0110">

Reply:
Yeah, and with my comment I suggested add small assertion to check if it true for `NotFound` page

Reply:
> Yeah, and with my comment I suggested to add small assertion to check if it true for `NotFound` page

Please check the test again. The test check status code for the component with `variant` `h1` as you want

Reply:
> > Yeah, and with my comment I suggested to add small assertion to check if it true for `NotFound` page
> 
> Please check the test again. The test check status code with text for the component with `variant` `h1` as you want. Or what do you mean ?

Reply:
I mean adding following assertion. Also i suggest renaming `errorCode` to `errorCodeElement`:

```jsx
expect(errorCode.tagName).toBe("H1")
```

Reply:
> I mean adding following assertion. Also i suggest renaming `errorCode` to `errorCodeElement`:
> 
> ```js
> expect(errorCode.tagName).toBe("H1")
> ```

Why I should to do so ? It's already cover this component with test..

Reply:
I don't force you, it is just my recommendation

Reply:
> I don't force you, it is just my recommendation

I would like to ask you. Which advantages it give us ? Testing element instead the text ??

Reply:
You agreed with me that we should have at least one `h1` tag per page, so what is the problem?

Reply:
> You agreed with me that we should have at least one `h1` tag per page, so what is the problem?

I don't see the reason change test `getByText` to get `h1` with `tagName`

Reply:
You don't have to change `getByText`. Adding extra assertion is not a problem

Reply:
> You don't have to change `getByText`. Adding extra assertion is not a problem

I won't to add it. because my test will be enough

Reply:
> > You don't have to change `getByText`. Adding extra assertion is not a problem
> 
> I won't to add it. because my test will be enough and not to follow your recommendations... why ?
It changes smth ?